In [11]:
from math import factorial
from math import exp

# Домашняя работа по уроку 2. Дискретные распределения вероятностей

---------

## Задача 1

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. 

Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

### Решение

- знаем точную вероятность одного выстрела

- ищем вероятность наступления события 85 раз из 100

- проводится небольшое число испытаний, 100 раз

Это биномаильное распределение

Используем формулу Бернулли:

P(X = k) = C<sub>n</sub><sup>k</sup> * p<sup>k</sup> * q<sup>n - k</sup>

P(X = 85) = C<sub>100</sub><sup>85</sup> * 0.8<sup>85</sup> * 0.2<sup>15</sup>

In [12]:
f'Ответ: P(X) = {round((factorial(100) / (factorial(85) * factorial(15))) * (0.8 ** 85) * (0.2 ** 15) * 100, 2)}%.'

'Ответ: P(X) = 4.81%.'

---------

## Задача 2

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. 

В жилом комплексе после ремонта в один день включили 5000 новых лампочек. 

Какова вероятность, что ни одна из них не перегорит в первый день? 

Какова вероятность, что перегорят ровно две?

### Решение

- известна вероятность наступления события

- большое количество испытаний

- вероятность наступления события очень маленькая

Это распределение Пуассона

Используем формулу:

P<sub>m</sub> ~ ((λ<sup>m</sup>) / m!) * (e<sup>-λ</sup>)

n = 5000

p = 0.0004

λ = n * p = 5000 * 0.0004 = 2

m1 = 0

m2 = 2

- вероятность, что ни одна не перегорит:

P<sub>0</sub> ~ ((2<sup>0</sup>) / 0!) * (e<sup>-2</sup>)

In [13]:
f'P = {round(exp(-2) * 100, 2)}%'

'P = 13.53%'

- вероятность, что перегорят ровно 2 лампочки:

P<sub>2</sub> ~ ((2<sup>2</sup>) / 2!) * (e<sup>-2</sup>)

In [14]:
f'P = {round(2 * exp(-2) * 100, 2)}%'

'P = 27.07%'

Ответ: 13.53%, 27.07%

---------

## Задача 3

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

### Решение

- знаем точную вероятность выпадения орла - 0.5

- ищем вероятность наступления события 70 раз из 144

- проводится небольшое число испытаний, 144 раза

Это биномаильное распределение

Используем формулу Бернулли:

P(X = k) = C<sub>n</sub><sup>k</sup> * p<sup>k</sup> * q<sup>n - k</sup>

P(X = 70) = C<sub>144</sub><sup>70</sup> * 0.5<sup>70</sup> * 0.5<sup>144 - 70</sup>

In [15]:
f'Ответ: P(X) = {round((factorial(144) / (factorial(70) * factorial(74))) * (0.5 ** 70) * (0.5 ** 74) * 100, 2)}%.'

'Ответ: P(X) = 6.28%.'

---------

## Задача 4

В первом ящике находится 10 мячей, из которых 7 - белые. 

Во втором ящике - 11 мячей, из которых 9 белых. 

Из каждого ящика вытаскивают случайным образом по два мяча. 

1) Какова вероятность того, что все мячи белые? 

2) Какова вероятность того, что ровно два мяча белые? 

3) Какова вероятность того, что хотя бы один мяч белый?

### Решение

1) Только один благоприятный исход: (ББ) (ББ)

P = (7 / 10) * (6 / 9) * (9 / 11) * (8 / 10)

In [16]:
f'P = {round((7/10) * (6/9) * (9/11) * (8/10) * 100, 2)}%'

'P = 30.55%'

2) 6 благоприятных исходов:

(ББ)(ЧЧ) + (БЧ)(БЧ) + (БЧ)(ЧБ) + (ЧБ)(БЧ) + (ЧБ)(ЧБ) + (ЧЧ)(ББ)

In [17]:
f'P = {round(((7/10)*(6/9)*(2/11)*(1/10))+((7/10)*(3/9)*(9/11)*(2/10))+((7/10)*(3/9)*(2/11)*(9/10))+((3/10)*(7/9)*(9/11)*(2/10))+((3/10)*(7/9)*(2/11)*(9/10))+((3/10)*(2/9)*(9/11)*(8/10)), 2)}%'

'P = 0.2%'

3) Данную задачу решим от обратного.

Найдем вероятность, что вытащат все черные мячи (ЧЧ)(ЧЧ)

И вычтем полученное из 1

In [18]:
f'P = {round((1 - ((3/10) * (2/9) * (2/11) * (1/10))) * 100, 2)}%'

'P = 99.88%'

Ответ: 1) 30,55% 2) 0,2% 3) 99,88%.